### DISCUSSION TITLE
machine-config degraded after installation

### DISCUSSION THREAD

**Discussion by sven-borkert** [okd-project/okd#1396] [Q&A]
Hi,

I did an installation with user provisioned infrastructure, according to this documentation:

https://docs.okd.io/latest/installing/installing_platform_agnostic/installing-platform-agnostic.html

Installer reports version: openshift-install 4.11.0-0.okd-2022-11-05-030711

The installation seems to have run without visible problems, but one of the cluster operators does not become happy when I run "watch -n5 oc get clusteroperators":

machine-config                                                              True        True          True       47m     Unable to apply 4.11.0-0.okd-2022-11-05-030711: error during syncRequiredMachineConfigPools: [timed out waiting for the condition, error pool master is not ready, retrying. Status: (pool degraded: true total: 3, ready 0, updated: 0, unavailable: 3)]

Cluster settings show that it is trying to update the master nodes, but cannot because machine config pools are degraded:

![grafik](https://user-images.githubusercontent.com/10104966/200681643-5a3d865b-e7d2-4303-93e2-fd7cff611c87.png)

$ oc get mcp
NAME     CONFIG                                             UPDATED   UPDATING   DEGRADED   MACHINECOUNT   READYMACHINECOUNT   UPDATEDMACHINECOUNT   DEGRADEDMACHINECOUNT   AGE
master                                                      False     True       True       3              0                   0                     3                      69m
worker   rendered-worker-86b9dd586f014e19df80cd351a231f83   True      False      False      2              2                   2                     0                      69m

$ oc get machineconfigs
NAME                                                    GENERATEDBYCONTROLLER                      IGNITIONVERSION   AGE
00-master                                               0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             71m
00-worker                                               0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             71m
01-master-container-runtime                             0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             71m
01-master-kubelet                                       0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             71m
01-worker-container-runtime                             0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             71m
01-worker-kubelet                                       0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             71m
99-master-generated-crio-add-inheritable-capabilities                                              3.2.0             71m
99-master-generated-registries                          0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             71m
99-master-okd-extensions                                                                           3.2.0             82m
99-master-ssh                                                                                      3.2.0             82m
99-worker-generated-crio-add-inheritable-capabilities                                              3.2.0             71m
99-worker-generated-registries                          0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             71m
99-worker-ssh                                                                                      3.2.0             82m
rendered-master-c8e03707c9787d303379616eca44f8ad        0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             71m
rendered-worker-86b9dd586f014e19df80cd351a231f83        0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             71m

The list of pods shows a pod "installer-6-master2.ocp4.borkert.net" has failed. The only visible error in the log of that pod is:

F1108 20:37:05.089596       1 cmd.go:106] Get "https://172.30.0.1:443/api/v1/namespaces/openshift-kube-scheduler/pods?labelSelector=app%3Dinstaller": dial tcp 172.30.0.1:443: connect: connection refused

Machine config pool "master" in the console shows this error:

![grafik](https://user-images.githubusercontent.com/10104966/200684182-dee21b43-922a-4b0f-ae3d-21953abf19bd.png)

Could someone please give me a hint what is wrong with the system, or what I have to read to understand this. I could not yet find anything helpful.

Regards,
Sven

**Comment by vrutkovs** [comment]
Dupe of #963 possibly?

**Reply by sven-borkert** [reply]
Yes, #963 seems to be about the same issue. I just don't see any solution in there? You commited 2 fixes in the end, but that is already months again, but these don't seem to have solved it for me.

I'm using the installer you released 3 days ago:

openshift-install 4.11.0-0.okd-2022-11-05-030711
built from commit 9d1e216a2ac5fede74450d1383c990ecc97c59da
release image quay.io/openshift/okd@sha256:d37ea780f755e8cda5ba6989f43060d1d2fccd1a98f9106bda70a307d1e04d40
release architecture amd64

I'm using "coreos-installer 0.16.1" with the generated ignition files from the rescue system of my servers to install the systems:

./coreos-installer install -i bootstrap.ign /dev/sda
...etc

And I'm using the example configuration file from https://docs.okd.io/latest/installing/installing_platform_agnostic/installing-platform-agnostic.html#installation-initializing-manual_installing-platform-agnostic where I added my SSH key and the pull secret. 

So I'm pretty much doing a basic default installation, what do I have to change to get rid of that errors? I don't see any solution in the referenced ticket.

Regards,
Sven

**Reply by vrutkovs** [reply]
Hopefully https://github.com/openshift/installer/pull/6488 (and its cherrypicks) would fix that

**Reply by sven-borkert** [reply]
Is the described procedure of deleting these 2 machine configs a valid workaround for now?

oc delete mc 99-master-okd-extensions 99-okd-master-disable-mitigations

I'd just like to have my environment running in a somewhat usable state, so I can run some examples from a book that I'm currently reading.

**Reply by quarterturn** [reply]
I had the same issue. running the "oc delete" commands above seems to have cleared it. "oc get clusteroperators" shows: machine-config                                                              True        True          False      10h     Working towards 4.11.0-0.okd-2022-11-05-030711
After a minute or two I now have everything available and nothing failed.

**Reply by sven-borkert** [reply]
I deleted "99-master-okd-extensions", but there is no "99-okd-master-disable-mitigations" on my cluster. 

After some minutes the mc I deleted just re-appeared, and there error is still present so far:

```
% oc get mc
NAME                                                    GENERATEDBYCONTROLLER                      IGNITIONVERSION   AGE
00-master                                               0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             18h
00-worker                                               0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             18h
01-master-container-runtime                             0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             18h
01-master-kubelet                                       0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             18h
01-worker-container-runtime                             0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             18h
01-worker-kubelet                                       0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             18h
99-master-generated-crio-add-inheritable-capabilities                                              3.2.0             18h
99-master-generated-registries                          0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             18h
99-master-okd-extensions                                                                           3.2.0             2m53s
99-master-ssh                                                                                      3.2.0             19h
99-worker-generated-crio-add-inheritable-capabilities                                              3.2.0             18h
99-worker-generated-registries                          0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             18h
99-worker-ssh                                                                                      3.2.0             19h
rendered-master-c8e03707c9787d303379616eca44f8ad        0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             18h
rendered-worker-86b9dd586f014e19df80cd351a231f83        0854b1512e8e445c235252a76e42043bbfa67512   3.2.0             18h

```
```
% oc get clusteroperators
NAME                                       VERSION                          AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
authentication                             4.11.0-0.okd-2022-11-05-030711   True        False         False      23m     
cloud-controller-manager                   4.11.0-0.okd-2022-11-05-030711   True        False         False      19h     
cloud-credential                           4.11.0-0.okd-2022-11-05-030711   True        False         False      19h     
cluster-autoscaler                         4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
config-operator                            4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
console                                    4.11.0-0.okd-2022-11-05-030711   True        False         False      23m     
csi-snapshot-controller                    4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
dns                                        4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
etcd                                       4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
image-registry                             4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
ingress                                    4.11.0-0.okd-2022-11-05-030711   True        False         False      23m     
insights                                   4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
kube-apiserver                             4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
kube-controller-manager                    4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
kube-scheduler                             4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
kube-storage-version-migrator              4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
machine-api                                4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
machine-approver                           4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
machine-config                                                              True        True          True       18h     Unable to apply 4.11.0-0.okd-2022-11-05-030711: error during syncRequiredMachineConfigPools: [timed out waiting for the condition, error pool master is not ready, retrying. Status: (pool degraded: true total: 3, ready 0, updated: 0, unavailable: 3)]
monitoring                                 4.11.0-0.okd-2022-11-05-030711   True        False         False      23m     
network                                    4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
node-tuning                                4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
openshift-apiserver                        4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
openshift-controller-manager               4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
openshift-samples                          4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
operator-lifecycle-manager                 4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
operator-lifecycle-manager-catalog         4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
operator-lifecycle-manager-packageserver   4.11.0-0.okd-2022-11-05-030711   True        False         False      24m     
service-ca                                 4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
storage                                    4.11.0-0.okd-2022-11-05-030711   True        False         False      18h     
```
Is it supposed to just re-appear after I deleted? How can I find out what else I need to remove to get the error fixed?

Regards,
Sven

**Reply by sven-borkert** [reply]
As deleting "99-master-okd-extensions" did not seem to have any effect, and there is no "99-okd-master-disable-mitigations" to delete, I tried something different:

The machine config pool says it wants to update:

```
      message: >-
        All nodes are updating to
        rendered-master-c8e03707c9787d303379616eca44f8ad
```
But it complains a machine configuration with a different ID is not existing:
```
master1.ocp4.borkert.net is reporting:
        "machineconfig.machineconfiguration.openshift.io
        \"rendered-master-7f562475a37ea947102e9b86ad46e14b\" not found"
```
I guess this is the machine config it things is the old one that it wants to update?

So in the hope to get the updater happy, I created "rendered-master-7f562475a37ea947102e9b86ad46e14b" as a copy of "rendered-master-c8e03707c9787d303379616eca44f8ad", so that it can find what it is searching for. 

This caused the errors to disappear, the mcp is now shown with no degraded machines anymore, and the clusteroperator also shows no error anymore.

![grafik](https://user-images.githubusercontent.com/10104966/200918498-2ba3951e-a221-401c-a29a-c0712939d2d5.png)

I hoped it would be fixed with this, but when I check the "Cluster Settings" it shows all green, but still "Failing":

![grafik](https://user-images.githubusercontent.com/10104966/200918768-baffd0e0-0b08-406e-b4f5-0052230262a5.png)

It seems to be missing some extensions now, it complains here:

![grafik](https://user-images.githubusercontent.com/10104966/200919039-11b88ad0-3caa-4d37-8dde-ecdde36476cb.png)

Any idea what is the problem now?

Regards,
Sven
